spec.bs

```diff
@@ -0,0 +1,520 @@
+
+  Note: In Chrome’s experimentation phase, the [=host=] of a {{Document}}'s {{URL}} is used as the [=determine-topics-calculation-input-data-header/topics calculation input data=], and the model is trained with human curated hostnames and topics.
+
+  <h2 id="collect-page-topics-calculation-input-data-header">Collect page topics calculation input data</h2>
+  <div algorithm>
+    To <dfn>collect page topics calculation input data</dfn>, given a {{Document}} |document|:
+
+    1. If |document|'s [=node navigable=] is a [=prerendering navigable=], then append the following steps to |document|'s [=post-prerendering activation steps list=] and return. Else, run the following steps [=in parallel=]:
+      1. Let |documentId| be |document|'s [=document-id-header/document ID=].
+      1. If user agent's [=user agent/topics history storage=] contains a [=topics history entry=] whose [=topics history entry/document id=] is |documentId|, return.
+      1. Let |topicsHistoryEntry| be a [=topics history entry=].
+      1. Set |topicsHistoryEntry|'s [=topics history entry/document id=] to |documentId|.
+      1. Set |topicsHistoryEntry|'s [=topics history entry/topics calculation input data=] to the [=determine-topics-calculation-input-data-header/topics calculation input data=] for |document|.
+      1. Set |topicsHistoryEntry|'s [=topics history entry/time=] to |document|'s [=relevant settings object=]'s [=current relative timestamp=].
+      1. [=list/Append=] |topicsHistoryEntry| to user agent's [=user agent/topics history storage=].
+  </div>
+
+  <h2 id="collect-topics-caller-origin-header">Collect topics caller origin</h2>
+  <div algorithm>
+    To <dfn>collect topics caller origin</dfn>, given a {{Document}} |document| and an [=origin=] |callerOrigin|:
+
+    1. Run the following steps [=in parallel=]:
+      1. Let |documentId| be |document|'s [=document-id-header/document ID=].
+      1. If user agent's [=user agent/topics history storage=] does not contain a [=topics history entry=] whose [=topics history entry/document id=] is |documentId|, return.
+      1. Let |topicsHistoryEntry| be the [=topics history entry=] in user agent's [=user agent/topics history storage=] whose [=topics history entry/document id=] is |documentId|.
+      1. [=set/Append=] |callerOrigin| to |topicsHistoryEntry|'s [=topics caller origins=].
+  </div>
+
+  <h2 id="periodically-calculate-user-topics-header">Periodically calculate user topics</h2>
+  <div algorithm>
+    To <dfn>calculate user topics</dfn>, perform the following steps every 7 days:
+    1. Let |now| be the [=monotonic clock=]'s [=monotonic clock/unsafe current time=].
```

Comment:
You probably want to use the wall clock here, since the monotonic clock isn't monotonic across browser restarts, and 7 days is a long time to expect no browser restarts. Then be sure to handle the case where the user sets their time backwards by a month or two.

Reply:
Done.